### PR TITLE
Isolate UWP builds from external gstreamers

### LINKED ISSUE #24327: UWP builds should set PKG_CONFIG_PATH
UWP builds use a local gstreamer from msvc-dependencies. They should override the existing PKG_CONFIG_PATH during the build so that the build doesn't go looking for x86 binaries

I _suspect_ this is why my older build was failing at dynamic link time, I had `PKG_CONFIG_PATH` set to x86 gstreamer, and uninstalling gstreamer fixed the problem but now the build fails when it builds glib-sys since it can't find glib. I'm not sure how it worked before since it was a clean build the first time, but it's definitely broken now. (My temporary fix was to just set `PKG_CONFIG_PATH` globally, it may be worth setting the `GSTREAMER_1_0_ROOT` variables instead)

cc @jdm

### PR DESCRIPTION
**Pull request by Manishearth**
Fixes https://github.com/servo/servo/issues/24327

This further seals the UWP build from the non-cross regular environment and makes pkg-config look at the UWP package.

This makes sure external gstreamer stuff doesn't sneak in. We already kinda do this by setting the LIB environment variable, but PKG_CONFIG_PATH also sneaks in and causes problems. Thing is, it turns out that the pkgconfig in gstreamer-uwp isn't enough for a full servo build, since we don't package gstreamer-webrtc, and the gstreamer-webrtc crate requires it to be around, even if we won't end up loading the library at runtime. Stuff has succeeded so far because people have gstreamer installations whose PKG_CONFIG_PATH is pulled in, despite us using a different set of DLLs, which somehow works but sometimes doesn't (I still don't know why).

I've added a fake gstreamer-webrtc-1.0.pc file to both targets in the gstreamer-uwp package with the following contents. It doesn't do anything the other pc files don't, so it doesn't end up pulling in additional libraries, it just exists to convince pkgconfig that we *have* this library (even though we don't), so that the build may succeed (we'll fail at runtime when we try to open WebRTC connections, but those are disabled anyway).

<details>

```pkgconfig

prefix=c:/gstreamer/1.0/arm64.uwp-release
exec_prefix=${prefix}
libdir=${prefix}/lib
includedir=${prefix}/include/gstreamer-1.0
datarootdir=${prefix}/share
datadir=${datarootdir}
girdir=${datadir}/gir-1.0
typelibdir=${libdir}/girepository-1.0

Name: Fake GStreamer WebRTC library
Description: Hacky stand-in for gstreamer-webrtc, does nothing but stops the Rust gstreamer-webrtc crate from failing during build
Requires: gstreamer-1.0 gstreamer-base-1.0
Version: 1.16.0
Libs: -L${libdir}
Cflags: -I${includedir}

```
</details>

r? @jdm

Perhaps don't merge yet, I can't test this until tomorrow (and ideally would like to test this from scratch tomorrow evening)

---
This change is [Reviewable](https://reviewable.io/reviews/servo/servo/24343)

### CODE REVIEW

**Review by jdm — APPROVED** [review]

**Comment by Manishearth** [comment]
@paul given that this is about fixing a "works on my machine" bug it would
be good if you could verify if this works on *your* machine as well.

File: python/servo/build_commands.py [hunk 1]
```diff
@@ -298,13 +298,17 @@ def build(self, release=False, dev=False, jobs=None, params=None,
...
+                "lib", "pkgconfig"
+            )
 
         # Ensure that GStreamer libraries are accessible when linking.
         if 'windows' in target_triple:
             gst_root = gstreamer_root(target_triple, env)
             if gst_root:
                 append_to_path_env(os.path.join(gst_root, "lib"), env, "LIB")
-
+
```
Comment (CYBAI) [inline python/servo/build_commands.py]: Found redundant spaces 👀

**Comment by ferjm** [comment]
Could we generate the fake `gstreamer-webrtc-1.0.pc` from Servo so we don't need to remember to add it when generating new binaries?

**Comment by Manishearth** [comment]
I'd rather document this since this is temporary: is there a place where we document how this package is generated?

**Comment by ferjm** [comment]
Ok.
There is no documentation yet. We have been using https://nirbheek.in/files/mozilla/uwp/gstreamer/0.4/ binaries so far. But these are missing the EGL related symbols that we need, so I am preparing a new version of the binaries (I got the x64 version so far, trying to figure out the cross compilation part). I will document the process when done and I'll add the fake webrtc bits there.

**Comment by paulrouget** [comment]
Builds and runs well here.

**Comment by jdm** [comment]
This can merge when you're happy with it.

**Pull request merged**